flasc/utilities/floris_tools.py

```diff
@@ -503,6 +507,88 @@ def add_gaussian_blending_to_floris_approx_table(df_fi_approx, wd_std=3.0, pdf_c
     return df_fi_approx_gauss
 
 
+def estimate_ws_with_floris(
+    df_scada: Union[pd.DataFrame, FlascDataFrame],
+    fm: FlorisModel,
+    verbose: bool = False,
+) -> Union[pd.DataFrame, FlascDataFrame]:
+    """Estimate the wind speed at the turbine locations using a FLORIS model.
+
+    This function estimates the wind speed at the turbine locations using a FLORIS model.
+    The approach follows the example from the RES wind-up method `add_ws_est_one_ttype`
+    (https://github.com/resgroup/wind-up/blob/main/wind_up/ws_est.py) by Alex Clerc.
+    However, in this implementation, FLORIS provides the power curves directly rather
+    than their being learned from data.  In this way, the estimated wind speed is
+    the speed which would cause FLORIS to predict a power matched to the SCADA data.
+    This will be especially useful in model fitting to avoid any error on un-waked turbines.
+
+    Args:
+        df_scada (pd.DataFrame | FlascDataFrame): Pandas DataFrame with the SCADA data.
```

Comment:
That's true, I think we only recommend it.  I guess it will fail very badly at least, but we could think about adding some tests that power is in kW.  Any values in excess of 20,000 would be a clue?

Reply:
Yeah, I'm trying to think whether any other procedures in FLASC assume that power is in kW. Possibly not, since we are often dividing a power by a reference power, so the units cancel out. But maybe there are other places on floris_tools.py where we make this assumption.